src/main/java/com/yelp/nrtsearch/server/luceneserver/field/LongFieldDef.java

```diff
@@ -118,4 +118,8 @@ public Query getTermQuery(TermQuery termQuery) {
   public Query getTermInSetQuery(TermInSetQuery termInSetQuery) {
     return LongPoint.newSetQuery(getName(), termInSetQuery.getLongTerms().getTermsList());
   }
+
+  protected Number parseNumberString(String numberString) {
+    return DOUBLE_PARSER.apply(numberString).longValue();
```

Comment:
Could this lead to precision loss for very large long values? Maybe do this if contains '.'?